src/main/java/bio/terra/buffer/service/resource/FlightManager.java

```
@@ -69,6 +86,29 @@ public FlightManager(
     return submitToStairway(flightSubmissionFactory.getCreationFlightSubmission(pool, resourceId));
   }
 
+  /**
+   * Create entity in resource table with CREATING and submit creation flight.
+   *
+   * <p>This should be done as a part of a transaction because we don't want resource state update
+   * without submitting a flight. The TransactionStatus is unused, but a part of the signature as a
+   * reminder.
+   *
+   * TODO: Remove before submit.
+   */
+  private Optional<String> createResourceEntityAndSubmitFlightWithTryCatch(
+      Pool pool, TransactionStatus unused) {
+    ResourceId resourceId = ResourceId.create(UUID.randomUUID());
+    bufferDao.createResource(
+        Resource.builder()
+            .id(resourceId)
+            .poolId(pool.id())
+            .creation(Instant.now())
+            .state(ResourceState.CREATING)
+            .build());
+    return submitToStairwayWithTryCatch(
+        flightSubmissionFactory.getCreationFlightSubmission(pool, resourceId));
```

If the flight submission has a try-catch, then we won't roll back the transaction if the flight submission fails. In that case, we'll have a CREATING resource sitting around without a Flight - effectively leaking the resource I think?

Oh, it is leaking, and the test verifies that... Am I supposed to be seeing with the two ways, that one doesn't work and one does? I'm ready to review the code that we're trying to submit.

---

All the TODO _withTryCatch is to demonstrate my thought(see my PR description).
After you and Greg reviewed the old leaking version with test, now I am removing those

---

Oops. Sorry. From the description "After digging into logs. I suspect TranscationTemplate doesn't propagate exceptions, and ignore exceptions being catched." I don't know what you mean by this. It doesn't magically propagate exceptions that don't bubble up to it? If the exception is caught within the TransactionCallback, then yes, it will not cause a rollback.
Also, we can't rollback stairway submissions with a TransactionTemplate - TransactionTemplate doesn't know anything about Stairway.

---

(1) I was just trying to say I saw transcation was leaked
(2) Feels that is a general problems of using TransactionTemplate, we don't really know how the sub-steam libraries we use implemented. But we use different libraries all the time. 
And to me using TransactionTemplate was more like to make sure our own db get rollbacked if something went wrong. 
And actually, I suspect Stairway can rollback as the code to TransactionTemplate is just another DB(in stairway DB) commit eventually

---

2) In general, you should not do anything other than query your db during a transaction. No API calls, no calls to other services, including Stairway. You can make exceptions, but carefully.
Today, Stairway cannot roll back as a part of the Spring TransactionTemplate. Stairway is connecting to a different database and isn't using any of the Spring magic. I'ts not obvious to me we'd want to support allowing its modifications to be undone as a part of a transaction.
We can verify this is the case:
```
String flightId = "foo"
transactionTemplate.execute(status -> { stairway.submitFlight(...); throw new RuntimeException(); });
stairway.getFlight(flightId);  // Does the flight submission exist even though the transaction was rolled back?
```

---

Yeah, we need to have the extra check step like I the one I added